examples/geometry/volumes/fullConvexitySphereGeodesics.cpp

```diff
@@ -0,0 +1,253 @@
+sphere, outputs the maximal distance (so less than pi), and checks
+that the furthest point is indeed antipodal to the source point. Two
+OBJ files named "sphere1-geodesics.obj "and
+"sphere1-geodesics-iso.obj" are also outputed, and you may use them to
+render the result like below.
+
+<table>
+<tr><td>
+\image html sphere1-h0_0625-geodesics.jpg "Exact geodesic distances on unit sphere digitized at h=0.0625 (max d=3.02389)" width=90%
+</td><td>
+\image html sphere1-h0_01-geodesics.jpg "Approximate geodesic distances on unit sphere digitized at h=0.01 (max d=3.1269)" width=90%
+</td></tr>
+</table>
+
+\example geometry/volumes/fullConvexitySphereGeodesics.cpp
+*/
+
+
+///////////////////////////////////////////////////////////////////////////////
+#include <iostream>
+#include <queue>
+#include "DGtal/base/Common.h"
+#include "DGtal/shapes/Shapes.h"
+#include "DGtal/shapes/SurfaceMesh.h"
+#include "DGtal/helpers/StdDefs.h"
+#include "DGtal/helpers/Shortcuts.h"
+#include "DGtal/images/ImageContainerBySTLVector.h"
+#include "DGtal/io/writers/SurfaceMeshWriter.h"
+#include "DGtal/geometry/volumes/TangencyComputer.h"
+
+///////////////////////////////////////////////////////////////////////////////
+
+using namespace std;
+using namespace DGtal;
+typedef Z3i::Space          Space;
+typedef Z3i::KSpace         KSpace;
+typedef Z3i::Domain         Domain;
+typedef Z3i::SCell          SCell;
+typedef Shortcuts< KSpace > SH3;
+typedef Space::Point        Point;
+typedef Space::RealPoint    RealPoint;
+typedef Space::RealVector   RealVector;
+typedef Space::Vector       Vector;
+typedef SurfaceMesh< RealPoint, RealVector > SMesh;
+typedef SurfaceMeshWriter< RealPoint, RealVector > SMeshWriter;
+typedef SMesh::Index        Index;
+typedef SMesh::Vertices     Vertices;
+
+void saveToObj( const std::string& output,
+                const SMesh& surfmesh,
+                const std::vector< double >& vvalues,
+                int nb_isolines_per_unit = 10,
+                const double thickness = 0.1 )
+{
+  std::string cobjname = output;
+  std::string cisoname = output + "-iso";
+  auto quantify = [] ( double v, double m, double nb )
+    { return round( v/m*nb )*m/nb; };
+  trace.beginBlock( "Output Corrected HD OBJ files" );
+  const auto fvalues = surfmesh.computeFaceValuesFromVertexValues( vvalues );
+  double maxValue = * ( std::max_element( vvalues.cbegin(), vvalues.cend() ) );
+  double minValue = * ( std::min_element( vvalues.cbegin(), vvalues.cend() ) );
+  double maxDist  = maxValue - minValue;
+  trace.info() << "Max distance is " << maxDist << std::endl;
+  auto cmap = SH3::getColorMap( 0.0, maxDist );
+  std::vector< Color > fcolors( surfmesh.nbFaces() );
+  for ( Index f = 0; f < fvalues.size(); ++f )
+    fcolors[ f ] = cmap( quantify( fvalues[ f ] - minValue, maxDist, 50 ) );
+  SMeshWriter::writeOBJ( cobjname, surfmesh, fcolors );
+  double unit = 1.0;
+  while ( unit < 0.01 * maxDist )        unit *= 10.0;
```

Comment:
can't a simple log do the job here instead of a while loop?

Reply:
Yes. Not so simple, but done ;-)